netket/optimizer/qgt/qgt_jacobian_dense.py

```diff
@@ -63,28 +63,45 @@ def QGTJacobianDense(
               models. holomorphic works for any function assuming it's holomorphic
               or real valued.
         holomorphic: a flag to indicate that the function is holomorphic.
-        diag_shift: Constant shift added to diagonal entries.
-        diag_scale: Fractional shift added to diagonal entries.
+        diag_scale: Fractional shift :math:`\epsilon_1` added to diagonal entries (see below)
+        diag_shift: Constant shift :math:`\epsilon_2` added to diagonal entries (see below)
         chunk_size: If supplied, overrides the chunk size of the variational state
                     (useful for models where the backward pass requires more
                     memory than the forward pass).
-    """
-    if mode is not None and holomorphic is not None:
-        raise ValueError("Cannot specify both `mode` and `holomorphic`.")
 
-    diag_shift, diag_scale = sanitize_diag_shift(diag_shift, diag_scale, rescale_shift)
+    Numerical estimates of the QGT are usually ill-conditioned and require
+    regularisation. The standard approach is to add a positive constant to the diagonal;
+    alternatively, Becca and Sorella (2017) propose scaling this offset with the
+    diagonal entry itself. NetKet allows using both in tandem:
+
+    .. math::
+
+        S_{ii} \mapsto S_{ii} + \epsilon_1 S_{ii} + \epsilon_2;
 
+    :math:`\epsilon_{1,2}` are specified using `diag_scale` and `diag_shift`,
+    respectively. By default, `diag_scale=0.0`, `diag_shift=0.01`.
+    """
     if vstate is None:
+        if mode is not None and holomorphic is not None:
+            raise ValueError("Cannot specify both `mode` and `holomorphic`.")
+        if rescale_shift is not None and diag_scale is not None:
+            raise ValueError("Cannot specify both `rescale_shift` and `diag_scale`.")
+        if diag_shift is not None:
+            kwargs["diag_shift"] = diag_shift
+        if diag_scale is not None:
+            kwargs["diag_scale"] = diag_scale
+        if rescale_shift is not None:
+            kwargs["rescale_shift"] = rescale_shift
```

Comment:
I'm not necessarily against this (though it's a lot of logic) but I fail to see how this helps us down the line..

Reply:
see [here](https://github.com/netket/netket/pull/1352#issuecomment-1281473410)

Reply:
I think this might be unneeded... See the following:

``python
In [5]: def f(**kwargs):
   ...:      print(kwargs)
   ...:
In [8]: partial(f, ciao=1)
Out[8]: functools.partial(<function f at 0x108665630>, ciao=1)
In [9]: partial(f, ciao=1)(ciao=3)
{'ciao': 3}
```

So you can shadow already defined partial warts by passing them from the top level.
Therefore I think it would be ok to drop this chunk of `if is not None...` and simply chug all keyword arguments in the partial 
```suggestion
```

Reply:
this works because we now sanitise the arguments after the partial, not before, which I think is the correct thing to do.

Reply:
> So you can shadow already defined partial warts by passing them from the top level.

OK that's interesting